4.queueRange.js

```diff
@@ -29,16 +29,18 @@
 //     return this.array.shift()
 //   }
 // }
-// 
+//
 // Imagine the queue contains -145, 45, 11, 98
 // range(q) should return 243
 
 // Write your function here
 const range = (queue) => {
+  if (queue.isEmpty()) return;
+  let min = queue.peek();
+  while (!queue.isEmpty()) {
+    max = Math.max(min, queue.dequeue());
+  }
```

Comment:
This is close. We need to have a min and max variable. 
el = q.dequeue()
max = Math.max(max, el);
min = Math.min(min, el)